### PR TITLE
fix(blocks): prevent carrying over modifiers when creating empty blocks

### PR DESCRIPTION
**Pull request by remidej**
When creating a new line with the enter key, if there is no text in the new line (so if the cursor was at the end of the previous line), all the modifiers should be disabled in the new block.

If the cursor is in the middle of the a block however, the block the new created block should have the text that was to the right of the cursor, _and_ keep the enabled modifiers.

### CODE REVIEW

File: packages/core/admin/admin/src/content-manager/components/BlocksEditor/hooks/tests/useBlocksStore.test.js [hunk 1]
```diff
@@ -897,4 +897,118 @@ describe('useBlocksStore', () => {
     // Check if the cursor is positioned in the new line
     expect(cursorPositionAfterSecondEnter[0]).toEqual(cursorInitialPosition[0] + 1);
   });
+
+  it('disables modifiers when creating a new node with enter key in a paragraph', () => {
```
Comment (simotae14) [inline packages/core/admin/admin/src/content-manager/components/BlocksEditor/hooks/tests/useBlocksStore.test.js:901]: Can you create also a test to check if the modifiers are active when you click enter in between an italic text for example?

**Review by simotae14 — APPROVED** [review]
LGMT and tested, 💯

**Review by simotae14 — COMMENTED** [review]
Hey @remidej I just QA your solution, I am not sure if it is what we want, if I create a new line from inside the block or quote where the text was bold the new line has the modifier applied as you can see in this video

https://github.com/strapi/strapi/assets/2589748/7c055988-1a33-4218-a456-b04f54465b0c

**Comment by remidej** [comment]
@simotae14 for code blocks, I didn't include it because I discussed with Yannis and in the end we will disable modifiers entirely for them (I created a new ticket for it).

For quote blocks, I thought I had fixed it with my last commit, I can't reproduce the behavior you see now. Can you double check that you pulled all the changes?

**Review by simotae14 — APPROVED** [review]
LGTM and tested

**Pull request merged**